pkg/controllers/cspc-controller/sync.go

```diff
@@ -261,3 +275,137 @@ func (c *Controller) GetCSPIWithoutDeployment(cspc *cstor.CStorPoolCluster) ([]c
 	}
 	return cspiList, nil
 }
+
+/*
+syncCSPI propagates all the required changes from cspc to respective cspi.
+ToDo: Currently -- in every resync interval the sync is tried and this needs to be improved by queuing cspc
+only at times when it is required.
+ */
+
+func (pc *PoolConfig) syncCSPI(cspc *cstor.CStorPoolCluster) error {
+	cspiList, err := pc.Controller.GetCSPIListForCSPC(cspc)
+	if err != nil {
+		return errors.Wrapf(err, "failed to sync cspi(s) from its parent cspc %s", cspc.Name)
+	}
+	if len(cspiList.Items) == 0 {
+		return errors.Wrapf(err, "No cspi(s) found while trying to sync cspi(s) from its parent cspc %s", cspc.Name)
+	}
+
+	for _, cspi := range cspiList.Items {
+		cspi := cspi
+		err := pc.syncCSPIWithCSPC(cspc, &cspi)
+		if err != nil {
+			klog.Errorf("failed to sync cspi %s from its parent cspc %s: {%s}", cspi.Name, cspc.Name, err)
+		}
+	}
+	return nil
+}
+
+// syncCSPIWithCSPC syncs a cspi with the parent cspc and hence updates the cspi's corresponding
+// pool manager deployment if required.
+func (pc *PoolConfig) syncCSPIWithCSPC(cspc *cstor.CStorPoolCluster, cspi *cstor.CStorPoolInstance) error {
+	cspiCopy := cspi.DeepCopy()
+	klog.Infof("Syncing cspi %s from parent cspc %s", cspiCopy.Name, cspc.Name)
+	for _, poolSpec := range cspc.Spec.Pools {
+		if reflect.DeepEqual(poolSpec.NodeSelector, cspi.Spec.NodeSelector) {
+			poolSpec := poolSpec
+			cspiCopy.Spec.PoolConfig = poolSpec.PoolConfig
+			defaultPoolConfig(cspiCopy, cspc)
+			break
+		}
+	}
+	gotCSPI, err := pc.Controller.GetStoredCStorVersionClient().CStorPoolInstances(cspiCopy.Namespace).Update(cspiCopy)
```

Comment:
Should we need to do updating CSP and patching deployment only in below condition
```go
if !reflect.DeepEqual(cspi.Spec.PoolConfig, cspiCopy.Spec.PoolConfig) {
// Step to perform update
// Step to perform patching of deployment
}
```